bindings/twilio/sendgrid/sendgrid_test.go

```diff
@@ -14,36 +14,66 @@ limitations under the License.
 package sendgrid
 
 import (
+	"context"
 	"encoding/json"
+	"net/http"
+	"net/http/httptest"
 	"testing"
 
+	"github.com/sendgrid/sendgrid-go/helpers/mail"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
 	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/components-contrib/metadata"
 	"github.com/dapr/kit/logger"
 )
 
-func TestParseMetadata(t *testing.T) {
```

Comment:
I kept the same tests that were here before but I tested through the public `Init` function instead of the private functions.